Use "command -v" instead of "which"

The behavior of "which" is not standardized by POSIX. Some old
implementations print their error messages to stdout instead of stderr,
and don't return a nonzero exit code when they fail to find the given
program. "command -v" on the other hand is in POSIX (optional in 2004
and required as of 2008).

https://github.com/fletcher/MultiMarkdown-6/pull/150
--- scripts/markdown.orig
+++ scripts/markdown
@@ -10,8 +10,7 @@
 # Be sure to include multimarkdown in our PATH
 export PATH="$PWD:/usr/local/bin:$PATH"
 
-which multimarkdown > /dev/null
-if [ $? = 1 ]
+if ! command -v multimarkdown >/dev/null 2>&1
 then
 	echo multimarkdown executable not found! >&2
 	exit 1
--- scripts/mmd.orig
+++ scripts/mmd
@@ -10,8 +10,7 @@
 # Be sure to include multimarkdown in our PATH
 export PATH="$PWD:/usr/local/bin:$PATH"
 
-which multimarkdown > /dev/null
-if [ $? = 1 ]
+if ! command -v multimarkdown >/dev/null 2>&1
 then
 	echo multimarkdown executable not found! >&2
 	exit 1
--- scripts/mmd2all.orig
+++ scripts/mmd2all
@@ -11,8 +11,7 @@
 # Be sure to include multimarkdown in our PATH
 export PATH="$PWD:/usr/local/bin:$PATH"
 
-which multimarkdown > /dev/null
-if [ $? = 1 ]
+if ! command -v multimarkdown >/dev/null 2>&1
 then
 	echo multimarkdown executable not found! >&2
 	exit 1
--- scripts/mmd2epub.orig
+++ scripts/mmd2epub
@@ -10,8 +10,7 @@
 # Be sure to include multimarkdown in our PATH
 export PATH="$PWD:/usr/local/bin:$PATH"
 
-which multimarkdown > /dev/null
-if [ $? = 1 ]
+if ! command -v multimarkdown >/dev/null 2>&1
 then
 	echo multimarkdown executable not found! >&2
 	exit 1
--- scripts/mmd2fodt.orig
+++ scripts/mmd2fodt
@@ -10,8 +10,7 @@
 # Be sure to include multimarkdown in our PATH
 export PATH="$PWD:/usr/local/bin:$PATH"
 
-which multimarkdown > /dev/null
-if [ $? = 1 ]
+if ! command -v multimarkdown >/dev/null 2>&1
 then
 	echo multimarkdown executable not found! >&2
 	exit 1
--- scripts/mmd2odt.orig
+++ scripts/mmd2odt
@@ -10,8 +10,7 @@
 # Be sure to include multimarkdown in our PATH
 export PATH="$PWD:/usr/local/bin:$PATH"
 
-which multimarkdown > /dev/null
-if [ $? = 1 ]
+if ! command -v multimarkdown >/dev/null 2>&1
 then
 	echo multimarkdown executable not found! >&2
 	exit 1
--- scripts/mmd2opml.orig
+++ scripts/mmd2opml
@@ -10,8 +10,7 @@
 # Be sure to include multimarkdown in our PATH
 export PATH="$PWD:/usr/local/bin:$PATH"
 
-which multimarkdown > /dev/null
-if [ $? = 1 ]
+if ! command -v multimarkdown >/dev/null 2>&1
 then
 	echo multimarkdown executable not found! >&2
 	exit 1
--- scripts/mmd2pdf.orig
+++ scripts/mmd2pdf
@@ -17,8 +17,7 @@
 # Be sure to include multimarkdown and latex in our PATH
 export PATH="$PWD:/usr/local/bin:/usr/texbin:/Library/TeX/texbin:$PATH"
 
-which multimarkdown > /dev/null
-if [ $? = 1 ]
+if ! command -v multimarkdown >/dev/null 2>&1
 then
 	echo multimarkdown executable not found! >&2
 	exit 1
--- scripts/mmd2tex.orig
+++ scripts/mmd2tex
@@ -10,8 +10,7 @@
 # Be sure to include multimarkdown in our PATH
 export PATH="$PWD:/usr/local/bin:$PATH"
 
-which multimarkdown > /dev/null
-if [ $? = 1 ]
+if ! command -v multimarkdown >/dev/null 2>&1
 then
 	echo multimarkdown executable not found! >&2
 	exit 1
